### PR TITLE
Dit 514 Integration Test

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import uk.gov.hmrc.bindingtarifffilestore.model.FileMetadata
+import uk.gov.hmrc.bindingtarifffilestore.repository.FileMetadataMongoRepository
+
+import scala.concurrent.Await.result
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+
+abstract class BaseFeatureSpec extends FeatureSpec with Matchers with GivenWhenThen with GuiceOneServerPerSuite with BeforeAndAfterEach with BeforeAndAfterAll {
+
+  private val timeout = 2.seconds
```
Comment (gokyo) [inline test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala]: no need to define it
you can use `UnitSpec`

File: test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  override protected def afterAll(): Unit = {
+    drop()
+  }
+
+  private def drop(): Unit = {
+    result(store.drop, timeout)
+  }
+
+  protected def store(files: FileMetadata*): Seq[FileMetadata] = {
+    files.map(c => result(store.insert(c), timeout))
```
Comment (gokyo) [inline test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala]: You can use `Future.sequence`

File: app/uk/gov/hmrc/bindingtarifffilestore/connector/UpscanConnector.scala [hunk 3]
```diff
@@ -37,7 +37,11 @@ class UpscanConnector @Inject()(appConfig: AppConfig, http: HttpClient, ws: WSCl
 
   def initiate(uploadSettings: UploadSettings)
               (implicit headerCarrier: HeaderCarrier): Future[UpscanInitiateResponse] = {
-    http.POST[UploadSettings, UpscanInitiateResponse](s"${appConfig.upscanInitiateUrl}/upscan/initiate", uploadSettings)
+    http.POST[UploadSettings, UpscanInitiateResponse](
+      s"${appConfig.upscanInitiateUrl}/upscan/initiate",
+      uploadSettings,
+      headers = Seq("User-Agent" -> appConfig.appName)
```
Comment (gokyo) [inline app/uk/gov/hmrc/bindingtarifffilestore/connector/UpscanConnector.scala]: The `User-Agent` header is automatically injected by the http-verbs library
You can remove it from here

Comment (ed0906) [reply app/uk/gov/hmrc/bindingtarifffilestore/connector/UpscanConnector.scala]: We need to override it dont we?

File: test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala (hunk 2 above)
Comment (gokyo) [reply test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala]: `await(Future.sequence(files.map(store.insert)))`

File: app/uk/gov/hmrc/bindingtarifffilestore/connector/UpscanConnector.scala (hunk 3 above)
Comment (ed0906) [reply app/uk/gov/hmrc/bindingtarifffilestore/connector/UpscanConnector.scala]: Maybe not :)

File: test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  override protected def beforeEach(): Unit = {
+    drop()
+    ensureIndexes()
+  }
+
+  override protected def afterAll(): Unit = {
+    drop()
+  }
+
+  private def drop(): Unit = {
```
Comment (gokyo) [inline test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala]: The methods `drop` and `storeSize` are never used.
Do we need them later?

Comment (ed0906) [reply test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala]: Drop is used in that same class

Comment (ed0906) [reply test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala]: store size removed

File: test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala (hunk 2 above)
Comment (ed0906) [reply test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala]: Changed

File: test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    result(store.drop, timeout)
+  }
+
+  protected def store(files: FileMetadata*): Seq[FileMetadata] = {
```
Comment (gokyo) [inline test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala]: this is never used I believe

Comment (ed0906) [reply test/it/uk/gov/hmrc/bindingtarifffilestore/BaseFeatureSpec.scala]: Removed

**Pull request merged**